### PR TITLE
bugfix-Segmentaion fault in planner_server node. #1608

### PR DESCRIPTION
**Pull request by crthilakraj**
---

## Basic Info

| Info | Please fill out this column |
| ------ | ----------- |
| Ticket(s) this addresses   | #1608 |
| Primary OS tested on | Ubuntu|
| Robotic platform tested on | ( gazebo simulation of Turtlebot) |

### CODE REVIEW

File: nav2_navfn_planner/src/navfn_planner.cpp [hunk 1]
```diff
@@ -246,19 +246,20 @@ NavfnPlanner::smoothApproachToGoal(
...
-  if (
-    squared_distance(last_pose.pose, second_to_last_pose.pose) >
-    squared_distance(goal, second_to_last_pose.pose))
-  {
-    plan.poses.back().pose = goal;
-  } else {
-    geometry_msgs::msg::PoseStamped goal_copy;
-    goal_copy.pose = goal;
-    plan.poses.push_back(goal_copy);
+  if (plan.poses.size() > 2){
```
Comment (SteveMacenski) [inline nav2_navfn_planner/src/navfn_planner.cpp]:
```suggestion
  if (plan.poses.size() > 2) {
```

Comment (crthilakraj) [reply nav2_navfn_planner/src/navfn_planner.cpp]: Corrected the changes requested. Regarding retarget to master, does that mean the eloquent- devel will not have these changes? or master will be merged later to eloquent-devel

Comment (crthilakraj) [reply nav2_navfn_planner/src/navfn_planner.cpp]: Also, should I create a new pull request for the master branch with cherrypicked commits from the current pull request?

Comment (SteveMacenski) [reply nav2_navfn_planner/src/navfn_planner.cpp]: Master may be eventually synced with eloquent, but all new development goes to master. You may create a new PR or rebase.

Comment (crthilakraj) [reply nav2_navfn_planner/src/navfn_planner.cpp]: https://github.com/ros-planning/navigation2/pull/1612 have raised the new PR, we can close this PR

**Pull request closed without merging**